doc/release/1.8.0-notes.rst

```diff
@@ -50,80 +86,808 @@ New features
 `scipy.optimize` improvements
 =============================
 
+`scipy.optimize.check_grad` introduces two new optional keyword only arguments, 
+``direction`` and ``seed``. ``direction`` can take values, ``'all'`` (default),
+in which case all the one hot direction vectors will be used for verifying
+the input analytical gradient function and ``'random'``, in which case a
+random direction vector will be used for the same purpose. ``seed``
+(default is ``None``) can be used for reproducing the return value of
+``check_grad`` function. It will be used only when ``direction='random'``.
+
+The `scipy.optimize.minimize` ``TNC`` method has been rewritten to use Cython
+bindings. This also fixes an issue with the callback altering the state of the
+optimization.
+
+Added optional parameters ``target_accept_rate`` and ``stepwise_factor`` for
+adapative step size adjustment in ``basinhopping``.
+
+The ``epsilon`` argument to ``approx_fprime`` is now optional so that it may
+have a default value consistent with most other functions in `scipy.optimize`.
 
 `scipy.signal` improvements
 ===========================
 
+Add ``analog`` argument, default ``False``, to ``zpk2sos``, and add new pairing
+option ``'minimal'`` to construct analog and minimal discrete SOS arrays.
+``tf2sos`` uses zpk2sos; add ``analog`` argument here as well, and pass it on
+to ``zpk2sos``.
+
+``savgol_coeffs`` and ``savgol_filter`` now work for even window lengths.
+
+Added the Chirp Z-transform and Zoom FFT available as `scipy.signal.CZT` and
+`scipy.signal.ZoomFFT`.
 
 `scipy.sparse` improvements
 ===========================
 
+An array API has been added to facilitate the ecosystem-wide transition away
+from the matrix API. We recommend that new work uses these array objects:
+``bsr_array``, ``coo_array``, etc. Please refer to the `scipy.sparse`
+docstring for more information.
+
+``maximum_flow`` introduces optional keyword only argument, ``method``
+which accepts either, ``'edmonds-karp'`` (Edmonds Karp algorithm) or
+``'dinic'`` (Dinic's algorithm). Moreover, ``'dinic'`` is used as default
+value for ``method`` which means that Dinic's algorithm is used for computing
+maximum flow unless specified. See, the comparison between the supported
+algorithms in
+`this comment <https://github.com/scipy/scipy/pull/14358#issue-684212523>`_.
+
+Parameters ``atol``, ``btol`` now default to 1e-6 in
+`scipy.sparse.linalg.isolve.lsmr` to match with default values in
```

Comment:
Change to `scipy.sparse.linalg.lsmr`, and the next line to `scipy.sparse.linalg.lsqr`.

Reply:
done, thx